contextual_orchestrator/provider_catalog.py

```
@@ -0,0 +1,960 @@
+        url = f"{agent.base_url.rstrip('/')}/models/v2/{model_path}"
+        request = urllib.request.Request(
+            url,
+            data=json.dumps({"input": messages}).encode("utf-8"),
+            headers={
+                "Authorization": f"Key {credential}",
+                "Accept": "application/json",
+                "Content-Type": "application/json",
+            },
+            method="POST",
+        )
+        with urllib.request.urlopen(request, timeout=self.timeout) as response:  # nosec B310
```

Ruff (0.16.1)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)